Jenkinsfile

```diff
@@ -0,0 +1,319 @@
+                      env.userSelModule='All'    
+                    }
+        
+                    println("User selected module " + env.userSelModule);
+                }
+            }
+        }
+        stage('Build maingateway-service') {
+            environment {
+                serviceName = 'maingateway-service'
+            }
+            when {
+                expression {
+                    env.userInput == 'Gateway' || env.userInput == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
+
+            }
+        }
+        stage('Build fisuser-service') {
+            environment { 
+                serviceName = 'fisuser-service'
+            }
+            when {
+                expression {
+                    env.userSelModule == 'FisUser' || env.userSelModule == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
+           }
+        }
+        stage('Build fisalert-service') {
+            environment { 
+                serviceName = 'fisalert-service'
+            }
+            when {
+                expression {
+                    env.userSelModule == 'FisAlert' || env.userSelModule == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
+            }
+        }
+        stage('Build nodejsalert-ui') {
+            environment { 
+                serviceName = 'nodejsalert-ui'
+            }
+            when {
+                expression {
+                    env.userSelModule == 'UI' || env.userSelModule == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName}"
+                node ('nodejs') {
+                    git params.GIT_REPO 
+ 
+                    script {
+                        sh """
+                        cd ${serviceName}
+
+                        oc project ${CICD_PROJECT}
+
+                        npm install && npm run openshift
+                        """
+                    }
+                } 
+            }
+        }
+
+        stage('Pushing to Test - maingateway') {
```

Comment:
cane we make a general stage to for push to image in different environment to remove the duplication of methods 
For example : stage("pushing image into ${app_name} into ${env_name)})

Reply:
For the stage, I would prefer to put them in a dedicated position as it is and not creating any separated method for them, as this could create code complexity and unnecessary logic inside/outside the method.